test/client/fixtures/sandbox/cookie-test.js

```diff
@@ -234,3 +208,21 @@ if (!browserUtils.isIOS) {
         nextCookieTest(0);
     });
 }
+
+asyncTest('limit of the failed cookie-sync messages (GH-1193)', function () {
+    var storedCookieSyncUrl = settings.get().cookieSyncUrl;
+
+    settings.get().cookieSyncUrl = '/cookie-sync-fail/';
+
+    setCookie('a=b');
+
+    setTimeout(function () {
+        strictEqual(cookieSync.failsCount, 3);
```

Comment:
Also need to override the `_onRequestError` and check that this is called 3 or more times